src/Core/View.js

```diff
@@ -175,47 +169,11 @@ function _preprocessLayer(view, layer, provider, parentLayer) {
         });
     }
 
-    // probably not the best place to do this
-    if (layer.type == 'color') {
-        defineLayerProperty(layer, 'frozen', false);
-        defineLayerProperty(layer, 'visible', true);
-        defineLayerProperty(layer, 'opacity', 1.0);
-        defineLayerProperty(layer, 'sequence', 0);
-    } else if (layer.type == 'elevation') {
-        defineLayerProperty(layer, 'frozen', false);
-    } else if (layer.type == 'geometry' || layer.type == 'debug') {
-        defineLayerProperty(layer, 'visible', true, () => _syncGeometryLayerVisibility(layer, view));
-        defineLayerProperty(layer, 'frozen', false);
+    if (layer.type == 'geometry' || layer.type == 'debug') {
+        layer.defineLayerProperty('visible', true, () => _syncGeometryLayerVisibility(layer, view));
```

Comment:
It's a shame that (layer.defineLayerProperty) this remains
It should use the notifyChange mechanism